doc/spec/namespace.md

```diff
@@ -0,0 +1,159 @@
+# Namespace
+The namespace represents the full set of named repositories which can be
+referenced by Docker. A repository contains a set of content-addressable blobs
+and tags referencing those blobs. The namespace can be used to discover the
+location and trust information of a repository based on DNS, HTTP requests and
+other means. The namespace is global and owned through the domain at the root of
+the namespace path.
+
+A repository should always be referenced by its fully canonical value. If an
+engine presents a shortened name to a user, that name should be fully
+canonicalized based on rules defined by the engine before contacting a remote
+server. When mirroring repositories, the original name for a repository must be
+used. Changing a repository name is equivalent to copying or moving and will not
+be referenced by the original repository.
+
```

Comment:
Definitions we need here:

- "scope": a namespace prefix, possibly a named repo itself.
- what characters can make up a namespace. The APPC spec reference the [DNS RFC](http://tools.ietf.org/html/rfc1123#page-13) to define the set of characters available for image names. We should have a look at that and see if it works. I think you can get part of this definition from the api spec.

Let's also add an example section:

1. Docker hub namespace: short name, prefixed by username/org (ie library/*, stevvooe/test, etc.)
2. Org namespaces (openshift/*, etc.)
3. Domain-prefixed namespaces. (docker.com/library/*, etc.)

Number 1 and 2 are what is already present. 3 is what we are trying to move to.